### PR TITLE
Point mbedtls module to zephyr/mbedTLS PR #38

### PR DESCRIPTION
**Pull request by alpsayin**
This will allow testing little-endian big number assembly multiplication fix at https://github.com/zephyrproject-rtos/mbedtls/pull/38

Signed-off-by: Alp Sayin <alpsayin@gmail.com>

### CODE REVIEW

**Comment by alpsayin** [comment]
Note to self: Compliance checks didn't like using commit number of the PR in west.yml or I was again unlucky with a network issue.

**Comment by alpsayin** [comment]
Depends on #53899 because https://github.com/zephyrproject-rtos/mbedtls/pull/38 is based on mbedTLS v3.2 which needs TFM v1.7

**Comment by joerchan** [comment]
Please rebase

**Comment by alpsayin** [comment]
> Please rebase

Rebase complete, all checks seem to pass.

**Review by ceolin — APPROVED** [review]
Once mbedTLS gets merged this is good to go IMHO.

**Review by de-nordic — COMMENTED** [review]
Shouldn't this be given Zephr bug issue and followed with release notes entry?

**Comment by alpsayin** [comment]
@de-nordic 
>  Shouldn't this be given Zephr bug issue and followed with release notes entry?

This is something I wouldn't know. Sounds like a maintainers/management decision.
Though logically, it also sounds like a Zephyr bug issue would first require microblaze support to be in Zephyr first.
I don't mind either way.

**Comment by alpsayin** [comment]
Original PR became unnecessary with mbedtls 3. 3.0 upgrade via https://github.com/zephyrproject-rtos/mbedtls/pull/42

**Pull request closed without merging**
